dash/include/dash/HaloMatrix.h

```diff
@@ -0,0 +1,259 @@
+#ifndef DASH__HALOMATRIX_H_INCLUDED
+#define DASH__HALOMATRIX_H_INCLUDED
+
+#include <dash/dart/if/dart.h>
+
+//#include <dash/Team.h>
+#include <dash/Pattern.h>
+//#include <dash/GlobRef.h>
+#include <dash/GlobMem.h>
+//#include <dash/Allocator.h>
+#include <dash/iterator/HaloMatrixIterator.h>
+#include <dash/Halo.h>
+#include <dash/Matrix.h>
+#include <type_traits>
+
+namespace dash {
+
+template<typename MatrixT, typename HaloSpecT>
+class HaloMatrix
```

Comment:
@fmoessbauer Yes, that was my intention with `dash::HaloBlock` exactly. It's just a view on a regular NArray block. `dash::HaloBlock` then again represents a sequence container.

See:
- HaloBlock: \
  https://github.com/dash-project/dash/blob/development/dash/include/dash/Halo.h#L1274
- BlockBoundaryView: \
  https://github.com/dash-project/dash/blob/development/dash/include/dash/Halo.h#L1064